File: sdkwork-ai-sdk-flutter/lib/src/api/videos2.dart
import '../http/client.dart';
import '../models.dart';

class Videos2Api {
  final HttpClient _client;
  
  Videos2Api(this._client);

  /// List videos
  Future<OpenAiVideoList?> listVideos(Map<String, dynamic>? params) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/videos'), params: params);
    return response is OpenAiVideoList ? response : null;
  }

  /// Create video
  Future<OpenAiVideo?> createVideo(VideoGenerationRequest body) async {
    final response = await _client.post(ApiPaths.aiPath('/v1/videos'), body: body, contentType: 'application/json');
    return response is OpenAiVideo ? response : null;
  }

  /// Retrieve video
  Future<OpenAiVideo?> retrieveVideo(String video_id) async {
    final response = await _client.get(ApiPaths.aiPath('/v1/videos/${video_id}'));
    return response is OpenAiVideo ? response : null;
  }

  /// Delete video
  Future<OpenAiVideoDeleteResponse?> deleteVideo(String video_id) async {
    final response = await _client.delete(ApiPaths.aiPath('/v1/videos/${video_id}'));
    return response is OpenAiVideoDeleteResponse ? response : null;
  }
}

File: sdkwork-ai-sdk-python/sdkwork_ai_sdk/api/videos2.py
from typing import Any, Dict, List, Optional
from ..http_client import HttpClient
from ..models import OpenAiVideo, OpenAiVideoDeleteResponse, OpenAiVideoList, VideoGenerationRequest

class Videos2Api:
    """Videos API client."""
    
    def __init__(self, client: HttpClient):
        self._client = client

    def list_videos(self, params: Optional[Dict[str, Any]] = None) -> OpenAiVideoList:
        """List videos"""
        return self._client.get(f"/ai/v3/v1/videos", params=params)

    def create_video(self, body: VideoGenerationRequest) -> OpenAiVideo:
        """Create video"""
        return self._client.post(f"/ai/v3/v1/videos", json=body)

    def retrieve_video(self, video_id: str) -> OpenAiVideo:
        """Retrieve video"""
        return self._client.get(f"/ai/v3/v1/videos/{video_id}")

    def delete_video(self, video_id: str) -> OpenAiVideoDeleteResponse:
        """Delete video"""
        return self._client.delete(f"/ai/v3/v1/videos/{video_id}")
